### PR TITLE
Revise docs/protocol/architecture

### PR DESCRIPTION
**Pull request by d-ashe**
### Description of the Change

Revision of docs/protocol/architecture.md to update for go-cosmwasm references.

### Release Notes

- Removed reference to secret-contracts not being on mainnet
- Changed references from 'compute' module to 'wasm' module
- Updated list of modules enabled on mainnet
- Updated link for cosmwasm CosmosMsg -> go-cosmwasm CosmosMsg

### CODE REVIEW

File: docs/protocol/architecture.md [hunk 1]
```diff
@@ -1,10 +1,8 @@
...
-
-A Secret Contract, written in Rust, is the fundamental innovation of the Secret Network. Secret Contracts are enabled by the `compute` module, which is unique to the Secret Network. These contracts execute over data which is kept encrypted from nodes, developers, users, and everyone else, while the results of these computations are trusted and verifiable. For application developers, the Secret Contract is the most important feature of the network.
+A Secret Contract, written in Rust, is the fundamental innovation of the Secret Network. Secret Contracts are enabled by the `wasm` module, which is unique to the Secret Network. These contracts execute over data which is kept encrypted from nodes, developers, users, and everyone else, while the results of these computations are trusted and verifiable. For application developers, the Secret Contract is the most important feature of the network.
```
Comment (assafmo) [inline docs/protocol/architecture.md:5]: The module is called compute and not wasm

File: docs/protocol/architecture.md [hunk 2]
```diff
@@ -24,17 +22,17 @@ The following process describes, step by step, how a contract is submitted and a
...
 
-This encrypted data can only be accessed from within the “trusted execution environment”, or enclave, that the `compute` module requires each validator to run. The computation indicated by the Secret Contract is then performed, within this trusted enclave, over the decrypted data. When the computation is completed, the output is encrypted and recorded on-chain. There are various types of outputs that can be expected. These include:
+This encrypted data can only be accessed from within the “trusted execution environment”, or enclave, that the `wasm` module requires each validator to run. The computation indicated by the Secret Contract is then performed, within this trusted enclave, over the decrypted data. When the computation is completed, the output is encrypted and recorded on-chain. There are various types of outputs that can be expected. These include:
```
Comment (assafmo) [inline docs/protocol/architecture.md:25]: Again, compute instead of wasm

File: docs/protocol/architecture.md [hunk 3]
```diff
@@ -24,17 +22,17 @@ The following process describes, step by step, how a contract is submitted and a
...
+- Send Messages to other modules (i.e., for sending value transfer messages that depend on the outcome of a computation). See [from go-cosmwasm code](https://github.com/enigmampc/SecretNetwork/blob/master/go-cosmwasm/types/msg.go#L63-L69)
 
 The Secret Network’s `compute` module currently requires that validators run nodes with Intel SGX chips (enclaves). These enclaves contain signing keys that are generated within the enclave. For more details on how enclaves function and are verified, see [intel SGX](sgx.md).
 
 ![enclave](../images/diagrams/enclave.png)
 
-Diagram: Trusted and Untrusted aspects of Secret Network code. `compute` enables cosmwasm with encryption to be executed within the trusted component of the enclave.
+Diagram: Trusted and Untrusted aspects of Secret Network code. `wasm` enables go-cosmwasm with encryption to be executed within the trusted component of the enclave.
```
Comment (assafmo) [inline docs/protocol/architecture.md:36]: Again, compute

Comment (d-ashe) [reply docs/protocol/architecture.md]: Good to know, thanks Assaf! I wasn't sure which one was canon.

**Review by assafmo — APPROVED** [review]
Thank you!

**Pull request merged**